File: 0080 Remove Duplicates from Sorted Array II/remove_duplicates_from_sorted_array_II.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if not nums:
            return 0;
        length = 0
        for num in nums:
            if length < 2 or num > nums[length - 2]:
                nums[length] = num
                length += 1
        return length

File: 0080 Remove Duplicates from Sorted Array II/RemoveDuplicatesFromSortedArrayII.java
class Solution {
    public int removeDuplicates(int[] nums) {
        if (nums == null || nums.length < 1) {
            return 0;
        }
        int len = 0;
        for (int num: nums) {
            if (len < 2 || num > nums[len - 2]) {
                nums[len] = num;    
                len++;
            }
        }
        return len;
    }
}